core/src/impl/Kokkos_Half_MathematicalFunctions.hpp

```diff
@@ -20,6 +20,18 @@
 #include <Kokkos_MathematicalFunctions.hpp>  // For the float overloads
 #include <Kokkos_BitManipulation.hpp>        // bit_cast
 
+// Backends specific half implementation needs to be declared prior to the math
+// functions implementation so that specifics functions can be found if they
+// exist.
+// If this includes are removed, this mechanism will fail silently.
```

Comment:
```suggestion
// Backend specific half implementation needs to be declared prior to the math
// functions implementation so that specific functions can be found if they
// exist.
// If these includes are removed, this mechanism will fall back to the general implementation.
```